d2/src/main/java/com/linkedin/d2/xds/XdsToD2PropertiesAdaptor.java

```diff
@@ -314,16 +330,29 @@ XdsClient.SymlinkNodeResourceWatcher getSymlinkResourceWatcher(String symlinkNam
       public void onChanged(String resourceName, XdsClient.NodeUpdate update)
       {
         // Update maps between symlink name and actual node name
-        String actualResourceName = update.getNodeData().getData().toString(StandardCharsets.UTF_8);
-        String actualNodeName = getNodeName(actualResourceName);
-        updateSymlinkAndActualNodeMap(symlinkName, actualNodeName);
-        // listen to the actual nodes
-        // Note: since cluster symlink and uri parent symlink always point to the same actual node name, and it's a
-        // redundancy and a burden for the symlink-update tool to maintain two symlinks for the same actual node name,
-        // we optimize here to use the cluster symlink to listen to the actual nodes for both cluster
-        // and uri parent.
-        listenToCluster(actualNodeName);
-        listenToUris(actualNodeName);
+        try
+        {
+          String actualResourceName = update.getNodeData().getData().toString(StandardCharsets.UTF_8);
+          String actualNodeName = getNodeName(actualResourceName);
+          updateSymlinkAndActualNodeMap(symlinkName, actualNodeName);
+          // listen to the actual nodes
+          // Note: since cluster symlink and uri parent symlink always point to the same actual node name, and it's a
+          // redundancy and a burden for the symlink-update tool to maintain two symlinks for the same actual node name,
+          // we optimize here to use the cluster symlink to listen to the actual nodes for both cluster
+          // and uri parent.
+          listenToCluster(actualNodeName);
+          listenToUris(actualNodeName);
+        }
+        catch (Exception e)
+        {
+          LOG.error("Failed to parse symlink data from xDS update. Symlink name: " + symlinkName, e);
```

Comment:
```suggestion
          LOG.error("Failed to parse symlink data from xDS update. Symlink name: {}", symlinkName, e);
```

Reply:
This is written by me. But I will change